docs/DeveloperGuide.adoc

```diff
@@ -831,35 +835,162 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |JCRC member |mass import residents' details |save time from not having to manually key in contacts one by one
+
+|`* * *` |JCRC member |delete selected groups of residents |save time from not having to search for and delete persons one by one
+
+|`* * *` |JCRC member |delete selected tags from residents |accommodate mass changes in CCA members list
+
+|`* * *` |JCRC member |compose emails |ease the process of sending emails to groups of hall residents
+
+|`* * *` |JCRC member |publish calenders |prevent clashes between events and inform hall residents of upcoming events
+
+|`* * *` |JCRC Finance Director |keep track of each CCA's budget |prevent problems during an audit
+
+|`* *` |JCRC member |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+
+|`* *` |JCRC member |list which residents are in specified CCAs |contact personnel in the CCA more efficiently
+
+|`*` |JCRC member |sort persons by name |locate a resident easily
+
+|`*` |JCRC member |store pictures under contact details |easily identify hall residents and distinguish between those with similar names
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
 |=======================================================================
 
 _{More to be added}_
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `Hallper` and the *Actor* is the `JCRC member`, unless specified otherwise)
+
+[discrete]
+=== Use case: Add person
+
+*MSS*
+
+1. JCRC member requests to add a person with specific details
+2. Hallper adds the person
++
+Use case ends.
 
 [discrete]
 === Use case: Delete person
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  JCRC member requests to list persons
+2.  Hallper shows a list of persons
+3.  JCRC member requests to delete a specific person in the list
+4.  Hallper deletes the person
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. Hallper shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: List persons
+
+*MSS*
+
+1. JCRC member requests to list persons
+2. Hallper shows list of all persons
++
+Use case ends.
+
+[discrete]
+=== Use case: Search tag
+
+*MSS*
+
+1. JCRC member requests to search by specific tag
+2. Hallper shows list of only persons with specified tag
++
+Use case ends.
+
+[discrete]
+===Use case: Find person
+
+*MSS*
+
+1. JCRC member requests to find person containing specific name
+2. Hallper shows list of persons with matching name
++
+Use case ends.
+
```

Comment:
If there is no person containing specific name, Hallper shows a message that "There is no one in the list called <name>"??? Or this is not required?

Reply:
Generally is it required to include the "error" message in developer guide?

Reply:
Nope, too specific